src/disk_io_thread.cpp

```diff
@@ -3302,6 +3288,27 @@ namespace libtorrent
 		TORRENT_ASSERT(m_magic == 0x1337);
 	}
 
+	int disk_io_thread::num_threads() const
+	{
+		return m_generic_threads.max_threads() + m_hash_threads.max_threads();
+	}
+
+	disk_io_thread::job_queue& disk_io_thread::queue_for_job(disk_io_job* j)
+	{
+		if (m_hash_threads.max_threads() > 0 && j->action == disk_io_job::hash)
```

Comment:
Yes, this is checking the max specifically for the hasher thread pool, not the overall max.